core/ValidationModel.js

```diff
@@ -0,0 +1,143 @@
+                    'constraint: ',
+                    fieldConstraint
+                );
+                return;
+            }
+
+            this.addReaction({
+                track: () => model[field],
+                run: () => this.validateFieldAsync(field)
+            });
+        });
+    }
+
+    /**
+     * Checks whether a field is valid in the current validation state.
+     *
+     * @param field The field to check
+     * @returns {boolean} True if the field is valid, false otherwise
+     */
+    isFieldValid(field) {
+        return !has(this.errors, field);
+    }
+
+    /**
+     * Returns all validation error messages in a flat list
+     * @returns {Array} array of error message strings
+     */
+    get allErrorMessages() {
+        return flatten(values(this.errors));
+    }
+
+    /**
+     * Returns all validation error messages for the specified fields
+     * @param fields {Array|String} the fields to get the error messages for
+     * @returns {Array} array of error message string
+     */
+    getFieldErrors(fields) {
+        const {errors} = this;
+        if (!errors) return [];
+
+        fields = castArray(fields);
+        return flatten(fields.map(field => errors[field] || []));
+    }
+
+    /**
+     * Validates all fields on the model which have constraints configured.
+     *
+     * @returns {Promise} A Promise that resolves to an object containing the current validation state
+     */
+    async validateAsync() {
+        return validate.async(this.model, this.constraints)
+            .then(() => {
+                this.setResult({});
+                return {isValid: this.isValid, result: this.errors};
+            })
+            .catch((errors) => {
+                this.setResult({...errors});
+                return {isValid: this.isValid, result: this.errors};
+            });
+    }
+
+    /**
+     * Validates a single field on the model against the constraints.
+     * Results of this single-field-validation will be merged into the current validation state.
+     *
+     * @param field the field on the model to validate
+     * @returns {Promise} A Promise that resolves to an object containing the current validation state
+     */
+    async validateFieldAsync(field) {
+        if (!has(this.constraints, field)) {
+            throw new Error(`No constraint defined for field ${field}. Constraints: `,
+                this.constraints);
+        }
+
+        return validate.async(this.model, pick(this.constraints, field))
+            .then(() => {
+                this.setResult(omit(this.errors, field));
+                return {isValid: this.isValid, result: this.errors};
+            })
+            .catch((errors) => {
+                this.setResult(assign({}, this.errors, errors));
+                return {isValid: this.isValid, result: this.errors};
+            });
+    }
+
+    //---------------------------------------------
+    // Implementation (internal)
```

Comment:
Sorry to nit-pick 😃 but the "internal" is pretty well implied by this standard little implementation comment block.